dxp/sennajs_integration.md

```diff
@@ -0,0 +1,117 @@
+
+Senna.js, introduces the following concepts:
+
+-   **Routes** Determine which path will be controlled by Senna.js and which Screen will handle the content.
+-   **Surface** A DOM element with an `id` attribute, which is where the content will be rendered.
+-   **Screen** A special type of route handler to handle the navigation request.
+
+To simulate a single page application, Senna.js [tracks all links](https://github.com/liferay/senna.js/blob/059022b737de0f0108b7ef3f06f14d7ade3baa94/src/app/App.js#L104) within the Surface. When a link is triggered, Senna.js controls the event by preventing the default behavior to prevent navigation from being made by the browser and then triggers the navigation process.
+
+### Navigation
+
+![](../images/sennajs-navigate.png)
+
+When navigation is made by Senna.js it creates a Screen and creates an element in the DOM next to the active Screen where the Screen can render the elements of the next route.
+
+1. Identifies whether the path satisfies any route
+2. Invoke the Screen that handles this route
+3. Prepares the surface and sequentially calls the life cycle of a Screen
+    - `load`
+    - `getTitle`
+    - `beforeUpdateHistoryPath`
+    - `beforeUpdateHistoryState`
+    - `getSurfaceContent`
+    - `evaluateStyles`
+    - `flip`
+    - `evaluateScripts`
+4. Synchronizes the scroll position if necessary
+5. Finalize navigation
+    - Activate screen
+    - Caches the Screen in memory
+
+The Screen can control the entire loading and extraction cycle of content resources to allocate in the DOM. [`HtmlScreen`](https://github.com/liferay/senna.js/blob/master/src/screen/HtmlScreen.js) is responsible for a good part of the manipulation and extraction of information from the content of the request and extends [`RequestScreen`](https://github.com/liferay/senna.js/blob/master/src/screen/RequestScreen.js) that takes care of setting up the request of the requested path.
+
+```
+HtmlScreen -> RequestScreen -> Screen
+```
+
+When the load happens all the returned content is allocated in a virtual document, Screen does not add directly to the DOM to not cause problems and for a better experience during the transition, the transition to the page is done only when everything is ready (style, scripts, and elements).
+
+### Resource tracking
+
+![](../images/sennajs-flip.png)
+
+Resource tracking happens when navigation is being made, this includes the `style` and `script` tags, Screen looks for those elements that are marked with the `data-senna-track` attribute within the content that is in a virtual document or in the DOM.
+
+There are two types of tracking for these attributes:
+
+-   **Permanent** These are resources that persist during navigation regardless of the route
+-   **Temporary** These are resources that are removed in the next route navigation
+
+> Any resource that has this attribute declared and is within the document's `body` will be moved to the document's `head`.
```

Comment:
How are elements without the `data-senna-track` attribute treated by default? ;)

Reply:
I think it's "permanent" but I'll look into the code and document it

Reply:
Yeah, if the element is in the `head` without the attribute it will be permanent Senna.js will not touch this element. But if the element is in `body` or specifically within the content of the page without the attribute, the behavior will be like a temporary.

Reply:
Thanks @matuzalemsteles I'll add that right now